### PR TITLE
[Feature] Adding dashboard link

### LINKED ISSUE #48: [Feature] Adding Project Insight link
We can also add a GitHub Hypercrx link to the projects ranking list to attract more Hypercrx users~

image

Or we can first show a dashboard to demostrate the metircs, such as:

image

and then, if a user want more detail information, he can use the Hypercrx to oberserve.

And the dashboard can be a building block both in OpenLeaderboard and Hypercrx~

### PR DESCRIPTION
**Pull request by andyhuang18**
**Related issues：**

- resolve #48 

**Details:**

在open leaderboard上增加了一列，用于跳转到DataEase的dashboard上：

image

### CODE REVIEW

**Comment by zhicheng-ning** [comment]
good job!

File: src/components/table.js [hunk 1]
```diff
@@ -58,6 +58,50 @@ const activityColumns = (object)=>[
...
+            title: t('大屏'),
+            dataIndex: 'name',
+            align:'center',
+            width: '20%',
+            //根据行号判断，前300的项目可点击大屏
+            render: function (text, row, index) {
+                //console.log('andy',row);
+                let [org_name,repo_name]=text.split("/");
+                //console.log('year',year);
+                //console.log('month',month);
```
Comment (tyn1998) [inline src/components/table.js:72]: Hi, @andyhuang18, I have some suggestions :)

1. It's better to remove useless code such as `console.log` before creating a PR

File: src/components/table.js [hunk 2]
```diff
@@ -58,6 +58,50 @@ const activityColumns = (object)=>[
...
+                        mymonth = '11';
+                        break;
+                    }
+                    default:
+                        mymonth='0'+month;
+                        break
+                }
+                //console.log('myyear',myyear);
+                //console.log('mymonth',mymonth);
+                let param='{\"org_name\":\"'+org_name+'\",\"repo_name\":\"'+repo_name+'\",\"t_month\":\"'+myyear+'-'+mymonth+'-01\"}';
```
Comment (tyn1998) [inline src/components/table.js:98]: 2. using [JS template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) to substitute `+` can make your life easier.

File: src/components/table.js [hunk 3]
```diff
@@ -113,6 +157,50 @@ const activityDetailColumns = (object)=>[
...
+                        break
+                }
+                //console.log(myyear);
+                //console.log(mymonth);
+                let param='{\"org_name\":\"'+org_name+'\",\"repo_name\":\"'+repo_name+'\",\"t_month\":\"'+myyear+'-'+mymonth+'-01\"}';
+                //console.log(param);
+                param=window.btoa(param);
+                //console.log(param);
+                return <a href={"http://dataease.nzcer.cn/link/LeMILNSw?attachParams="+param} target="_blank">{repo_name}大屏看板</a>
+            }
```
Comment (tyn1998) [inline src/components/table.js:202]: 3. this code snippet repeats several times in this file, better extract it to a function then call the function any where~

**Review by tyn1998 — COMMENTED** [review]
👍

**Comment by tyn1998** [comment]
image

**Comment by andyhuang18** [comment]
Thanks for the suggestions! ^ ^

**Comment by wj23027** [comment]
Modified the problem mentioned by @tyn1998, and changed the dashboard link to an icon~
image

File: src/components/table.js [hunk 4]
```diff
@@ -227,6 +257,18 @@ const solveDate = (year,month)=>{
     }
     return year+"年"+(month+1)+"月";
 }
+
+function dashboard(text,index,t_month,region) {
+    if (index < 400 && region =='chinese' ) {
+        let [org_name, repo_name] = text.split("/");
+        let param = `{"org_name":"${org_name}","repo_name":"${repo_name}","t_month":"${t_month}"}`
+        param = window.btoa(param);
```
Comment (tyn1998) [inline src/components/table.js:265]: image

This piece of code can be improved by `JSON.stringfy()`.

File: src/components/table.js [hunk 5]
```diff
@@ -238,7 +280,7 @@ function MyTable(props){
         object:'company',
         index:'activity',
         region:'chinese',
-        columns: activityColumns('company'),
+        //columns: activityColumns('company'),
```
Comment (tyn1998) [inline src/components/table.js:283]: Useless comments should be removed :)

File: src/components/table.js [hunk 6]
```diff
@@ -311,7 +361,7 @@ function MyTable(props){
         else{
             url += year + (1+month)+ '.json';
         }
-        console.log(url);
+        console.log('url',url);
```
Comment (tyn1998) [inline src/components/table.js:364]: and here.

**Review by tyn1998 — CHANGES_REQUESTED** [review]
Great job @wj23027. I tested this PR locally and it works fine:

image

But the code can still be improved a bit :)

File: src/components/table.js (hunk 4 above)
Comment (wj23027) [reply src/components/table.js:265]: What a great method, I never knew that before, thank you for the suggestion!

**Review by tyn1998 — APPROVED** [review]
LGTM! Thank you @andyhuang18 @wj23027 🌷

**Pull request merged**